proto/workspace.proto

```diff
@@ -0,0 +1,140 @@
+syntax = "proto3";
+
+import "proto/context.proto";
+
+package workspace;
+
+message GetWorkspaceRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The name of the workspace to fetch.
+  string name = 2;
+}
+
+message GetWorkspaceResponse {
+  // The response context.
+  context.ResponseContext response_context = 1;
+
+  // The requested workspace.
+  // To read workspace file contents, use the `GetWorkspaceFile` endpoint.
+  // To read workspace directory contents, use the
+  // `GetWorkspaceDirectoryRequest` endpoint.
+  Workspace workspace = 2;
+}
+
+message SaveWorkspaceRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The workspace to save.
+  Workspace workspace = 2;
+}
+
+message SaveWorkspaceResponse {
+  // The response context.
+  context.ResponseContext response_context = 1;
+}
+
+message GetWorkspaceDirectoryRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // The name of the workspace from which to fetch this directory.
+  string workspace = 2;
+
+  // Information about the git repo from which to fetch this directory.
+  Repo repo = 3;
+
+  // The details about which directory to fetch.
+  Node directory = 4;
+}
+
+message GetWorkspaceDirectoryResponse {
+  // The response context.
+  context.ResponseContext response_context = 1;
+
+  // The directory that was requested.
+  Node directory = 3;
+
+  // The child nodes (files and directories) contained in this directory.
+  repeated Node child_nodes = 2;
```

Comment:
I'm cool with starting with this, but we might want to watch out b/c GetTree which is similar, has to be streaming to accommodate the flattened list of files under a parent directory.

Reply:
Specified this in the comment, but since this is just the first level of children - I'm hoping we won't have to do any streaming.